File: code/babylon.py
def babylonian_sqrt(S):

    if S == 0:
        return 0

    err = 10**(-10)

    init_guess = 20
    xn = init_guess

    while (abs(xn**2 - S)/S) > err:
        next_guess = 1/2 * (xn + S/xn)
        xn = next_guess

    return xn

print(babylonian_sqrt(5))

File: code/babylon.c
#include <stdio.h>
#include <math.h>
#include <stdlib.h>

double babylonian_sqrt(double S) {
	double next_guess, init_guess = 20;
	double xn = init_guess;

	double err = pow(10, -10);
	while ((abs(xn*xn - S) / S) > err) {
		next_guess = 1/2 * (xn + S/xn);
		xn = next_guess;
	}

	return xn;
}

int main() {
	double sq, S = 2;
	sq = babylonian_sqrt(S);
	printf("%f", sq);
	return 0;

}
